st2actions/st2actions/container/base.py

```diff
@@ -96,18 +96,15 @@ def _do_run(self, runner, runnertype_db, action_db, actionexec_db):
             runner.pre_run()
 
             LOG.debug('Performing run for runner: %s', runner)
-            if isinstance(runner, AsyncActionRunner):
-                (status, result, query_context) = runner.run(action_params)
-            else:
-                (status, result) = runner.run(action_params)
+            (status, result, context) = runner.run(action_params)
```

Comment:
The context contains the reference information for the mistral execution.  The query context is build by the mistral query module anyway and the context already contains the information that it needs to build the query string.  Now I can add context and query_context.  But I think it's even more confusing.  The query_context is not adding any new/different information.  So why not just delegate the runner to return appropriate context information and then let the query module build the query context it needs to do the query.  Also, I personally do not like the style that plugin can return different number of items in the tuple.